### PR TITLE
Add tutorials for how to connect with redux

### PR DESCRIPTION
**Pull request by PorterNan**
# What

Add tutorials for how to connect with redux

There is a bug for the current npm package that it can't work without react dependencies:
https://skype.visualstudio.com/SPOOL/_sprints/taskboard/Web%20UI/SPOOL/CY2021-Q3/Sprint%2041%20(Ends%20July%2025)?workitem=2523882

Once this bug gets fixed, will migrate this doc into our public storybook

# Why

# How Tested

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

**Review by JamesBurnside — APPROVED** [review]

File: docs/architecture/ConnectToRedux.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
+## Connect Stateful Client to Redux
+
+Redux is a popular state management framework, it provides a centralized app state. You can easily integrate redux with all popular UI frameworks(React, Angular, VueJs). Developers can eaily integrate our stateful client with redux, and connect redux to their own exisiting framework.
```
Comment (alcail) [inline docs/architecture/ConnectToRedux.md:3]: Redux is a popular state management framework **that** provides a centralized app state. You can easily integrate redux with all popular UI frameworks **(React, Angular, VueJs)**. Developers can **easily** integrate our stateful client with redux, and connect redux to their own **existing** framework.

File: docs/architecture/ConnectToRedux.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+Here is a quick tutorial for how to connect stateful client state to redux store:
+### Install dependencies
+
+```
+npm install @azure/communication-react redux
+```
+
+### Create a reducer
+
+This reducer of redux help you update the centralized redux store with our client state.
```
Comment (alcail) [inline docs/architecture/ConnectToRedux.md:16]: This reducer of redux help you **updates** the centralized redux store with our client state.

File: docs/architecture/ConnectToRedux.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    console.log(store.getState());
+});
+
+chatThreadClient.sendMessage({content: 'Hello Redux!'});
+
+```
+
+### Connect redux with your own UI framework
+
+Now you have an redux store connected with our stateful client, any state update from stateful client will be dispatched to redux store immediately. You can either use [react-redux](https://react-redux.js.org/) or choose your own [binding library](https://redux.js.org/introduction/ecosystem#library-integration-and-bindings) to connect redux to your own UI.
```
Comment (alcail) [inline docs/architecture/ConnectToRedux.md:121]: Now you have **a** redux store connected with our stateful client, ...

**Review by alcail — APPROVED** [review]

File: docs/architecture/ConnectToRedux.md [hunk 4]
```diff
@@ -0,0 +1,121 @@
+## Connect Stateful Client to Redux
+
+Redux is a popular state management framework, it provides a centralized app state. You can easily integrate redux with all popular UI frameworks(React, Angular, VueJs). Developers can eaily integrate our stateful client with redux, and connect redux to their own exisiting framework. 
+
+![Connect with redux](../images/stateful-redux-ui.png)
+
+Here is a quick tutorial for how to connect stateful client state to redux store:
```
Comment (anjulgarg) [inline docs/architecture/ConnectToRedux.md:7]: Here is a quick tutorial for connecting stateful client state to redux store:

File: docs/architecture/ConnectToRedux.md (hunk 2 above)
Comment (anjulgarg) [reply docs/architecture/ConnectToRedux.md:16]: This Redux reducer lets you update the centralized redux store with our client state.

**Review by anjulgarg — APPROVED** [review]

**Pull request merged**